src/lib.rs

```diff
@@ -191,70 +230,24 @@ impl fmt::Display for Filter {
     }
 }
 
-impl From<Filter> for LevelFilter {
-    fn from(filter: Filter) -> Self {
-        match filter {
-            Filter::Off => LevelFilter::Off,
-            Filter::Error => LevelFilter::Error,
-            Filter::Warn => LevelFilter::Warn,
-            Filter::Info => LevelFilter::Info,
-            Filter::Debug => LevelFilter::Debug,
-            Filter::Trace => LevelFilter::Trace,
-        }
-    }
-}
-
-impl From<LevelFilter> for Filter {
-    fn from(level: LevelFilter) -> Self {
-        match level {
-            LevelFilter::Off => Filter::Off,
-            LevelFilter::Error => Filter::Error,
-            LevelFilter::Warn => Filter::Warn,
-            LevelFilter::Info => Filter::Info,
-            LevelFilter::Debug => Filter::Debug,
-            LevelFilter::Trace => Filter::Trace,
-        }
-    }
-}
-
-impl From<Filter> for Option<Level> {
-    fn from(filter: Filter) -> Self {
-        match filter {
-            Filter::Off => None,
-            Filter::Error => Some(Level::Error),
-            Filter::Warn => Some(Level::Warn),
-            Filter::Info => Some(Level::Info),
-            Filter::Debug => Some(Level::Debug),
-            Filter::Trace => Some(Level::Trace),
-        }
-    }
-}
-
-impl From<Option<Level>> for Filter {
-    fn from(level: Option<Level>) -> Self {
-        match level {
-            None => Filter::Off,
-            Some(Level::Error) => Filter::Error,
-            Some(Level::Warn) => Filter::Warn,
-            Some(Level::Info) => Filter::Info,
-            Some(Level::Debug) => Filter::Debug,
-            Some(Level::Trace) => Filter::Trace,
-        }
-    }
-}
-
 use std::fmt;
 
-impl<L: LogLevel> fmt::Display for Verbosity<L> {
+impl<L: LogLevel> fmt::Display for Verbosity<L>
+where
+    Filter: Into<Option<L::Level>> + Into<L::LevelFilter> + From<Option<L::Level>>,
+{
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
         write!(f, "{}", self.filter())
     }
 }
 
 /// Customize the default log-level and associated help
 pub trait LogLevel {
+    type Level;
+    type LevelFilter;
```

Comment:
> no associated types used

With `Filter`, we shouldn't need any of this

Reply:
It was either use associated types, or create crate specific methods (e.g. `log_level`, `tracing_level`). I think the specific methods are a much simpler approach.

7b8b965